src/language-js/comments.js

```diff
@@ -563,7 +571,13 @@ function handleCommentInEmptyParens(text, enclosingNode, comment, options) {
       enclosingNode.type === "FunctionExpression" ||
       enclosingNode.type === "ArrowFunctionExpression" ||
       enclosingNode.type === "ClassMethod" ||
-      enclosingNode.type === "ObjectMethod") &&
+      enclosingNode.type === "ObjectMethod" ||
+      enclosingNode.type === "TSDeclareFunction" ||
+      enclosingNode.type === "TSCallSignatureDeclaration" ||
+      enclosingNode.type === "TSConstructSignatureDeclaration" ||
+      enclosingNode.type === "TSMethodSignature" ||
+      enclosingNode.type === "TSConstructorType" ||
+      enclosingNode.type === "TSFunctionType") &&
```

Comment:
Checking against the same list of types happens also in `handleLastFunctionArgComments`. Please extract this check to a function.

Reply:
do you have idea for name of this function?

Reply:
`hasSignature`

Reply:
or `isFunctionish`